pkg/kube/fake/fake.go

```diff
@@ -50,8 +51,11 @@ func (f *FailingKubeClient) Create(resources kube.ResourceList) (*kube.Result, e
 	return f.PrintingKubeClient.Create(resources)
 }
 
-// Wait returns the configured error if set or prints
+// Waits the amount of time defined on f.WaitDuration, then returns the configured error if set or prints.
 func (f *FailingKubeClient) Wait(resources kube.ResourceList, d time.Duration) error {
+	if f.WaitDuration != 0 {
```

Comment:
`time.Sleep(time.Duration(0 * time.Second))` is a no-op so this `if` check is redundant.

From https://pkg.go.dev/time#Sleep:

> A negative or zero duration causes Sleep to return immediately.

Reply:
Resolved in commit https://github.com/helm/helm/pull/9180/commits/4bc901c95f3589b6d6b4b3bc0843347334cd0f68